### PR TITLE
Tests, DELETE /api/job, code cleanup

### PR DESCRIPTION
**Pull request by 2bigO**
Coverage would be about 50% after this

### CODE REVIEW

**Comment by 2bigO** [comment]
Is it OK to move to node > 8?
I have extensively used `util.promisify`

File: package.json [hunk 1]
```diff
@@ -7,10 +7,10 @@
   "scripts": {
     "prepublish": "npm run build",
     "lint": "xo",
-    "dev": "npm run build && node $NODE_DEBUG_OPTION cli.js",
-    "build": "webpack",
-    "test": "ava && npm run lint",
-    "start": "npm run build && node cli.js"
+    "dev": "webpack --mode development",
+    "build": "webpack --mode production",
+    "test": "ava ./dist/test.js && npm run lint",
```
Comment (OmgImAlexis) [inline package.json]: Ah.. is the test transpiled? Ava already does this.

Comment (2bigO) [reply package.json]: Cool! Didn't know.

File: test.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
 const {promisify} = require('util');
 const test = require('ava');
 const request = require('supertest');
-const bootstrapKoaApp = require('./src/bootstrap-koa-app');
+const bootstrapKoaApp = require('./dist/bootstrap-koa-app');
```
Comment (OmgImAlexis) [inline test.js]: Could this not use the src as well? It transpiles everything within the test context.

Comment (2bigO) [reply test.js]: That threw an error hence the change. Also I have another problem with ava's transpilation. It doesn't use the plugins I specified. Or maybe it's not transpiling at all?

Comment (2bigO) [reply test.js]:
```
/home/keyvan/agenda-rest/src/bootstrap-koa-app.js:1
(function (exports, require, module, __filename, __dirname) { import Koa from 'koa';
                                                              ^^^^^^

SyntaxError: Unexpected token import
    at new Script (vm.js:51:7)
    at createScript (vm.js:138:10)
    at Object.runInThisContext (vm.js:199:10)
    at Module._compile (module.js:624:28)
    at Module._extensions..js (module.js:671:10)
    at extensions.(anonymous function) (/home/keyvan/agenda-rest/node_modules/require-precompiled/index.js:16:3)
    at Object.require.extensions.(anonymous function) [as .js] (/home/keyvan/agenda-rest/node_modules/ava/lib/process-adapter.js:100:4)
    at Module.load (module.js:573:32)
    at tryModuleLoad (module.js:513:12)
    at Function.Module._load (module.js:505:3)
    at Module.require (module.js:604:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/home/keyvan/agenda-rest/test.js:4:25)
    at Module._compile (module.js:660:30)
    at extensions.(anonymous function) (/home/keyvan/agenda-rest/node_modules/require-precompiled/index.js:13:11)
    at Object.require.extensions.(anonymous function) [as .js] (/home/keyvan/agenda-rest/node_modules/ava/lib/process-adapter.js:100:4)

  1 exception

  ✖ test.js exited with a non-zero exit code: 1
```

Comment (OmgImAlexis) [reply test.js]: Check [this page](https://github.com/avajs/ava/blob/master/docs/recipes/babel.md).

Comment (2bigO) [reply test.js]: Checking... thanks.

Comment (2bigO) [reply test.js]: After trying many things, I'm not able to get ava's transpilation to work properly. It's not reading my `.babelrc`, then even with the manual config (not even adding the plugins):
```json
"ava": {
    "babel": {
      "presets": [
        ["@babel/preset-env", {
          "targets": {
            "node": "current"
          }
        }]
      ]
    },
    "require": [
      "@babel/register"
    ]
  }
```

produces this error:

```
TypeError: Cannot read property 'loose' of undefined (While processing preset: "/home/keyvan/agenda-rest/node_modules/@babel/preset-env/lib/index.js")
    at _default (/home/keyvan/agenda-rest/node_modules/@babel/plugin-transform-modules-commonjs/lib/index.js:21:23)
    at Function.memoisePluginContainer (/home/keyvan/agenda-rest/node_modules/babel-core/lib/transformation/file/options/option-manager.js:113:13)
    at Function.normalisePlugin (/home/keyvan/agenda-rest/node_modules/babel-core/lib/transformation/file/options/option-manager.js:146:32)
    at /home/keyvan/agenda-rest/node_modules/babel-core/lib/transformation/file/options/option-manager.js:184:30
    at Array.map (<anonymous>)
    at Function.normalisePlugins (/home/keyvan/agenda-rest/node_modules/babel-core/lib/transformation/file/options/option-manager.js:158:20)
    at OptionManager.mergeOptions (/home/keyvan/agenda-rest/node_modules/babel-core/lib/transformation/file/options/option-manager.js:234:36)
    at /home/keyvan/agenda-rest/node_modules/babel-core/lib/transformation/file/options/option-manager.js:265:14
    at /home/keyvan/agenda-rest/node_modules/babel-core/lib/transformation/file/options/option-manager.js:323:22
    at Array.map (<anonymous>)
    at OptionManager.resolvePresets (/home/keyvan/agenda-rest/node_modules/babel-core/lib/transformation/file/options/option-manager.js:275:20)
    at OptionManager.mergePresets (/home/keyvan/agenda-rest/node_modules/babel-core/lib/transformation/file/options/option-manager.js:264:10)
    at OptionManager.mergeOptions (/home/keyvan/agenda-rest/node_modules/babel-core/lib/transformation/file/options/option-manager.js:249:14)
    at OptionManager.init (/home/keyvan/agenda-rest/node_modules/babel-core/lib/transformation/file/options/option-manager.js:368:12)
    at File.initOptions (/home/keyvan/agenda-rest/node_modules/babel-core/lib/transformation/file/index.js:212:65)
    at new File (/home/keyvan/agenda-rest/node_modules/babel-core/lib/transformation/file/index.js:135:24)
From previous event:
    at Api._runWithPool (/home/keyvan/agenda-rest/node_modules/ava/api.js:255:5)
    at _setupPrecompiler.then.then (/home/keyvan/agenda-rest/node_modules/ava/api.js:183:17)
    at <anonymous>
```

I'll use `require`s from `dist` until I find a solution.

On the plus side, babel is now at 7 (which was a requirement for ava's transpilation to work).

**Comment by 2bigO** [comment]
@OmgImAlexis 
Passing again. I'm done trying to get ava to transpile for now, see my comment.
You OK with the rest?

**Comment by OmgImAlexis** [comment]
Yeah, I'm not sure why you're having so many issues.

**Comment by 2bigO** [comment]
@OmgImAlexis  not many anymore, only ava's transpilation config, we can simply change:

```json
"test": "ava dist/test.js && npm run lint"
```

to

```json
"test": "ava && npm run lint"
```

when that is resolved. The requirements are now satisfied, only the config remains.

**Pull request merged**